bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java

```diff
@@ -0,0 +1,145 @@
+    connection.close();
+  }
+
+  @Test
+  public void testEstimatedAndSplitForSmallTable() throws Exception {
+    TableName tableName = newTestTableName();
+    try (Admin admin = connection.getAdmin()) {
+      admin.createTable(
+          new HTableDescriptor(tableName).addFamily(new HColumnDescriptor(COLUMN_FAMILY)));
+      try (Table table = connection.getTable(tableName)) {
+        table.put(Arrays.asList(
+          new Put(Bytes.toBytes("row1")).addColumn(COLUMN_FAMILY, QUALIFIER1, Bytes.toBytes("1")),
+          new Put(Bytes.toBytes("row2")).addColumn(COLUMN_FAMILY, QUALIFIER1, Bytes.toBytes("2"))));
+      }
+
+      List<SampleRowKeysResponse> sampleRowKeys = getSampleKeys(tableName);
+
+      try {
+        CloudBigtableIO.Source source =
+            new CloudBigtableIO.Source(new CloudBigtableScanConfiguration(
+                projectId, zoneId, clusterId, tableName.getQualifierAsString(), new Scan()));
+        long estimatedSizeBytes = source.getEstimatedSizeBytes(null);
+        SampleRowKeysResponse lastSample = sampleRowKeys.get(sampleRowKeys.size() - 1);
+        Assert.assertEquals(lastSample.getOffsetBytes(), estimatedSizeBytes);
+
+        List<SourceWithKeys> bundles = source.splitIntoBundles(estimatedSizeBytes / 2 + 1, null);
+        // This will be a small table with no splits, so we return HConstants.EMPTY_START_ROW
+        // which can't be split.
+        Assert.assertEquals(sampleRowKeys.size() * 2 - 1, bundles.size());
+      } finally {
+        admin.deleteTable(tableName);
+      }
+    }
+  }
+
+  @Test
+  public void testEstimatedAndSplitForLargeTable() throws Exception {
+    TableName tableName = newTestTableName();
+    try (Admin admin = connection.getAdmin()) {
+      admin.createTable(
+          new HTableDescriptor(tableName).addFamily(new HColumnDescriptor(COLUMN_FAMILY)));
+      try (BufferedMutator mutator = connection.getBufferedMutator(tableName)) {
+        for (int i = 0; i < 1000; i++ ) {
+          mutator.mutate(
+              new Put(Bytes.toBytes("row" + i))
+                  .addColumn(COLUMN_FAMILY, QUALIFIER1, getLargeValue()));
+        }
+      }
+
+      List<SampleRowKeysResponse> sampleRowKeys = getSampleKeys(tableName);
+
+      try {
+        CloudBigtableIO.Source source =
+            new CloudBigtableIO.Source(new CloudBigtableScanConfiguration(
+                projectId, zoneId, clusterId, tableName.getQualifierAsString(), new Scan()));
+        long estimatedSizeBytes = source.getEstimatedSizeBytes(null);
+        SampleRowKeysResponse lastSample = sampleRowKeys.get(sampleRowKeys.size() - 1);
+        Assert.assertEquals(lastSample.getOffsetBytes(), estimatedSizeBytes);
+
+        List<SourceWithKeys> bundles =
+            source.splitIntoBundles(sampleRowKeys.get(0).getOffsetBytes() / 2, null);
+        // The last sample includes the EMPTY_END_ROW key, which cannot be split.  
+        Assert.assertEquals(sampleRowKeys.size() * 2 - 1, bundles.size());
```

Comment:
It's difficult to figure out the right number of bundles, since it depends on the number of splits, which isn't easy to determine.  I added a check for key count.